### PR TITLE
fix: deprecate starlite integration. closes #3359

### LINKED ISSUE #3359: Deprecate starlite
Now that we Litestar support I think we can deprecate Starlite 😊

```[tasklist]
### Tasks
- [ ] Add notice in the docs
- [ ] Use typing_extensions.deprecated to mark class as deprecated (see https://peps.python.org/pep-0702/)
- [ ] Trigger deprecation warning (with test), this might 
```

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/strawberry-graphql) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/strawberry-graphql/strawberry/issues/3359">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/3359/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by Birdi7**
This PR deprecates the Starlite integration, as stated by [issue](https://github.com/strawberry-graphql/strawberry/issues/3359)

## Description
1. Mark `starlite.make_graphql_controller` function as deprecated
2. Mark `starlite.StarliteRequestAdapter` as deprecated class
3. Add test for both messages

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

Closes #3359

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

**Deprecations:** This release deprecates the `Starlite` integration in favour of the `LiteStar` integration.
Refer to the [LiteStar](./litestar.md) integration for more information.
LiteStar is a [renamed](https://litestar.dev/about/organization.html#litestar-and-starlite) and upgraded version of Starlite.

Before:

```python
from strawberry.starlite import make_graphql_controller
```

After:

```python
from strawberry.litestar import make_graphql_controller
```



Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Egor for the PR 👏

Get it here 👉 https://strawberry.rocks/release/(next)
```

File: strawberry/starlite/controller.py [hunk 1]
```diff
@@ -118,6 +119,10 @@ async def get_root_value(self) -> Any:
         return await self._get_root_value()
 
 
+@deprecated(
+    "The `starlite` integration is deprecated, please use `litestar` instead.",
+    stacklevel=2,
```
[the remark this answers is not in the document]
Comment (Birdi7) [reply strawberry/starlite/controller.py:124]: In examples inside `strawberry` I didn't find any deprecation with version to be removed in. So, nothing should be in this case

File: tests/starlite/test_context.py [hunk 2]
```diff
@@ -181,3 +183,22 @@ def something(self, info: strawberry.Info) -> str:
...
+def test_starlite_usage_triggers_deprecation_warning():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def abc(self, info: strawberry.Info) -> str:
+            assert info.context.get("request") is not None
+            assert info.context.get("strawberry") is None
+            return "abc"
+
+    schema = strawberry.Schema(query=Query)
```
[the remark this answers is not in the document]
Comment (Birdi7) [reply tests/starlite/test_context.py:197]: Yeap, added

**Comment by Birdi7** [comment]
@erikwrede would be great to get your review on this!

**Review by patrick91 — APPROVED** [review]
This is great! Thank you!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**